lib/routes/route.js

```diff
@@ -331,11 +340,66 @@ module.exports = function(app, router) {
                 logger.error(`Could not find image of ID ${id} for direct link, serving "removed" placeholder image via redirect`);
                 res.redirect("/images/removed.png");
             } else {
-                res.status(200);
-                res.append("Cache-Control", "public, max-age=2592000");
-                res.type(imageEntry[0].mimetype);
-                res.send(imageEntry[0].data.buffer); // Send the file data to the browser.
-                logger.info("Served image " + imageEntry[0].id + " by direct link");
+                if(req.query.res && req.query.res === "low") {
+                    var extension = req.params.ext;
+                    if(extension === "jpeg") {
```

Comment:
```suggestion
                    if (mimeType === "image/jpeg") {
```
Changing check to mime type will allow for clean check of both .jpg and .jpeg.